plugins/GSdx/GSVertexArrayOGL.h

```diff
@@ -161,21 +163,24 @@ class GSBufferOGL {
 			}
 		}
 
-		void* dst = m_buffer_ptr + offset;
+		return m_buffer_ptr + offset;
+	}
 
-		memcpy(dst, src, length);
-		glFlushMappedBufferRange(m_target, offset, length);
+	void unmap(size_t count)
+	{
+		m_count = count;
```

Comment:
```
unmap also takes a count, for cases where all the memory originally requested wasn't used.
```
I think it is a bad idea to do that. The buffer is a circular FIFO with sync object that is thread safe between driver and application. Code won't support well if you go back here.

If you really want to save some bandwidth, you can flush a smaller size. But you need to keep m_count as the real allocated size. The best behavior will be to reserve the exact size. Please note that you won't save memory as it is a circular buffer.

Note, potentially uses MAX_SIZE as default value. And do
`flush_count = std::min(count, m_count);`